HLTrigger/JetMET/plugins/HLTHFJetCleaner.cc

```diff
@@ -0,0 +1,269 @@
+
+  std::vector<float> v_sigmaEtaEta = extractValueMapParameters<float>(sigmaEtaEtas, jets->size());
+  std::vector<float> v_sigmaPhiPhi = extractValueMapParameters<float>(sigmaPhiPhis, jets->size());
+  std::vector<int> v_centralEtaStripSize = extractValueMapParameters<int>(centralEtaStripSizes, jets->size());
+
+  // Loop over the jets and do cleaning
+  for (size_t ijet = 0; ijet < jets->size(); ijet++) {
+    reco::Jet jet = (*jets)[ijet];
+    bool withinEtaRange = (abs(jet.eta()) < jetEtaMax_) & (abs(jet.eta()) > jetEtaMin_);
+    // Cuts are applied to jets that are back to back with MET
+    iEvent.getByToken(metToken_, mets);
+    const float met_phi = mets->front().phi();
+    bool backToBackWithMet = deltaPhi(jet.phi(), met_phi) > dphiJetMetThresh_;
+    if (!backToBackWithMet) {
+      cleaned_jets->push_back(jet);
+      continue;
+    }
+    // If the jet is outside the range that we're interested in,
+    // append it to the cleaned jets collection and continue
+    if ((jet.pt() < jetPtThresh_) | (!withinEtaRange)) {
+      cleaned_jets->push_back(jet);
+      continue;
+    }
+    // Cuts related to eta and phi width of HF showers:
+    // Sigma eta-phi based cut + central strip size cut
+    // We'll apply the sigma eta+phi dependent cut only if requested via the config file
+    if (applySigmaEtaPhiCut_) {
+      float sieie = v_sigmaEtaEta[ijet];
+      float sipip = v_sigmaPhiPhi[ijet];
+
+      // Sigma eta eta and sigma phi phi
+      bool passSigmaEtaPhiCut = sieie - sipip < sigmaEtaPhiDiffThresh_;
+
+      if (applyCornerCut_) {
+        bool inCorner = (v_sigmaEtaEta[ijet] < cornerCutSigmaEtaEta_) & (v_sigmaPhiPhi[ijet] < cornerCutSigmaPhiPhi_);
+        passSigmaEtaPhiCut = passSigmaEtaPhiCut & (!inCorner);
+      }
+      if (!passSigmaEtaPhiCut) {
+        continue;
+      }
+    }
+
+    if (applyStripSizeCut_) {
+      // Cut related to central strip size
+      int cess = v_centralEtaStripSize[ijet];
+      bool passCentralStripSizeCut = cess < centralEtaStripSizeThresh_;
+      if (!passCentralStripSizeCut) {
+        continue;
+      }
+    }
+
+    cleaned_jets->push_back(jet);
+  }
+
+  iEvent.put(std::move(cleaned_jets));
+}
+
+// ------------ method called when starting to processes a run  ------------
+/*
+void
+HLTHFJetCleaner::beginRun(edm::Run const&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method called when ending the processing of a run  ------------
+/*
+void
+HLTHFJetCleaner::endRun(edm::Run const&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method called when starting to processes a luminosity block  ------------
+/*
+void
+HLTHFJetCleaner::beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method called when ending the processing of a luminosity block  ------------
+/*
+void
+HLTHFJetCleaner::endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&)
+{
+}
+*/
+
```

Comment:
Remove commented-out blocks.

Reply:
Implemented in [65bcbcc](https://github.com/cms-sw/cmssw/pull/37539/commits/65bcbccb4d1e7fe0a8b6b2ea2d8835d03d8b6e99)